service/controller/resource/azureconfigfinalizer/resource.go

```diff
@@ -0,0 +1,103 @@
+package azureconfigfinalizer
+
+import (
+	"context"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "azureconfigfinalizer"
+
+	// Finalizer of old operator's controller.
+	legacyFinalizer = "operatorkit.giantswarm.io/azure-operator"
+)
+
+type Config struct {
+	CtrlClient client.Client
+	Logger     micrologger.Logger
+}
+
+// Resource manages the cluster ID label in the AzureConfig CR.
```

Comment:
![](https://media1.tenor.com/images/cf75220e2e89f3b23b686d4766088f2a/tenor.gif?itemid=15995669)

Reply:
![you didn't see anything](https://media.giphy.com/media/5zh1j8sUfLUJGI5T5d/giphy.gif)